### PR TITLE
Events Page, Details Page

### PR DESCRIPTION
**Pull request by awaisanwar544**
This PR includes these features:

1. Home page (Events Page)
2. Filter List of Events on the Home page
3. Detail Page for all the categories
4. All the pages have unique Route
5. Tests for all the reducers and components

Loom Video for Demo
https://www.loom.com/share/6f8ec70b5e4a4f009b6e36b11ba477c4

### CODE REVIEW

File: src/redux/eventsReducer.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import axios from 'axios';
+
+// globals
+const dict = {};
+const initialState = {};
+const URL = 'https://kontests.net/api/v1/all';
+
+// Actions
+const FETCH_EVENTS = 'FETCH_EVENTS';
```
Comment (FrederickMih) [inline src/redux/eventsReducer.js:9]: - [Optional] As a good practice, always try to scope the action names based on the feature they belong to. This helps when debugging more complex applications.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @awaisanwar544 , 

You have done a fantastic job so far. Keep up! :rocket: 
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

Screenshot from 2022-04-12 20-44-47

# To Highlights

-  Git/GitHub workflow has been followed :heavy_check_mark: 
-  Linters are correctly set up :heavy_check_mark: 
-  Pull Request has a title and summary :heavy_check_mark: 
- The design guidelines have been followed :+1: 

## Suggested changes

- [x] **Video presentation**: Hi @awaisanwar544 , the video presentation of your project is missing some requirements:
- The video should contain a clear and objective description of your project in professional English.
- The video should contain a demo of the project, highlighting all the project features. And,
- The video should contain a highlight about a piece of the code you find interesting or proud of making.
It will be nice if you can record another video to include the requirements above. You ended up presenting the features of the project on a browser. You didn't open up the IDE.

- [x] I think you are missing out on the filter stateless component to filter data retrieve from the API. From the requirements of this project, the `Home` page should have a list of categories that could be filtered using a given parameter, e,g, category name. Imagine introducing a word such as **Ha**... in a search field, it will retrieve all the events that begin with the letter **H** such as `HakerRank`, `HackerEarth`..... Please let's consider implementing this and move a step toward approval.
- [x] A link to the live version of your project in the README file is not working. Please let's fix it.
- [x] The `MIT` link of the License header of your README file is broken. Please let's also fix it.
- [x] The requirements of this project demand us to give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the author of the original design. Please let's do this in the `Acknowledgements `section of the  README file.
- [x] **Page structure**
Please let's separate our components into two: stateless components(placed in the component directory) and stateful components(placed in the container directory)
- [x] Please also make sure that your app is responsive so that mobile users can also browse without any difficulties.


_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/Events.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        onChange={filterHandler}
+      />
+      <ListWrapper>
+        {!filter && list.length > 0 && (
+          list.map((item) => <NavLink key={item} to={item.replace(/\s/g, '')}><EventCat category={item} /></NavLink>)
+        )}
+        {filter && list.length > 0 && (
+          list.filter((item) => (item.toLowerCase().includes(filter))).map((item) => <NavLink key={item} to={item.replace(/\s/g, '')}><EventCat category={item} /></NavLink>)
+        )}
+      </ListWrapper>
```
Comment (medaminedev66) [inline src/components/Events.jsx]: - [x] The filter search doesn't work as expected. When I try to search for the events that start with the `H` letter, the page doesn't give me the expected result (All the events' names that start with the `H` letter or contain the `H` letter). Kindly work on that so your project is complete.

*The search bar works well with lowercase letters. Let's make sure that the uppercase letters are detected in the search bar.*

image

File: src/components/Events.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    font-size: 2em;
+  }
+`;
+
+const ListWrapper = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  column-gap: 0.5em;
+  row-gap: 0.5em;
+`;
```
Comment (medaminedev66) [inline src/components/Events.jsx]: - [x] Your design still doesn't fit the mobile sizes. Let's improve it a bit so your design is responsive and fit the mobile screen sizes.
- Your design:
image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @awaisanwar544 😃,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- The repo looks professional ✔️
- There are no linter errors ✔️
- Your presentation is great ✔️
- Your app works as expected
- Your project uses Redux for state management and data retrieved from the API are stored in the Redux store correctly.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Events.jsx (hunk 2 above)
Comment (awaisanwar544) [reply src/components/Events.jsx]: resolved

File: src/components/Events.jsx (hunk 3 above)
Comment (awaisanwar544) [reply src/components/Events.jsx]: Screenshot 2022-04-13 at 7 17 34 AM

**Review by medaminedev66 — APPROVED** [review]
## Hi @awaisanwar544 😄, 

# Status: Approved ✔️
Your project is complete! 👍  There is nothing else to say other than... it's time to merge it 🚢
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**